frontend/src/components/dashboard/campaigns/announcement-modal/AnnouncementModal.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useContext, useState, useEffect } from 'react'
+import cx from 'classnames'
+import { updateAnnouncementVersion } from 'services/settings.service'
+
+import { PrimaryButton, ErrorBlock } from 'components/common'
+import { ModalContext } from 'contexts/modal.context'
+
+import styles from './AnnouncementModal.module.scss'
+import { i18n } from 'locales'
+import { ANNOUNCEMENT } from 'config'
+import { OutboundLink } from 'react-ga'
+
+const AnnouncementModal = () => {
+  const { close, setBeforeClose } = useContext(ModalContext)
+  const [errorMessage, setErrorMessage] = useState('')
+
+  useEffect(() => {
+    setBeforeClose(() => async () => {
+      try {
+        await updateAnnouncementVersion(i18n._(ANNOUNCEMENT.version))
+      } catch (err) {
+        setErrorMessage(err.message)
+      }
+    })
+    // eslint-disable-next-line
+  }, [])
```

Comment:
Any dependency we are trying to avoid here too? It's best practice to include dependencies suggested by `exhaustive-deps`, usually its a logic flaw on our side that may result in errors later on.

Reply:
Ah yes thanks for the catch, this is a change I should've reverted when trying to make the context work. Done!